projects/cm_mcu/LocalTasks.c

```diff
@@ -319,7 +334,7 @@ struct sm_command_t sm_command_fflot_f2[] = {
     {1, 0x00, 0x16, 2, "FF_TEMPERATURE", 0xff, "C", PM_STATUS},
     {2, 0x00, 0x07, 1, "FF_LOS_ALARM", 0xffff, "", PM_STATUS},
     {2, 0x00, 0x14, 1, "FF_CDR_LOL_ALARM", 0xffff, "", PM_STATUS},
-
+    {2, 0x01, 0xe4, 1, "FF_OPT_POW", 0xff, "mw", PM_STATUS}, // read 12 Rx-ch registers  with decreasing addresses
```

Comment:
same question

Reply:
another way to solve this problem is to use the 4th field which I think is the size of the data and make it something like 4 for the 4 channel devices and 12 for the 12 channel Rx device... (double check if that is the meaning of the 4th field ;-) )

Reply:
I was not sure what you meant by "the 4th field." Perhaps you meant sending output from every fourth channel? So, we would send 8 outputs for XCVR devices and 12 outputs for Rx devices.

Reply:
I mean the fourth field in each row, i.e., the `size` entry. see here: https://github.com/apollo-lhc/cm_mcu/blob/fd4fb8085d121b84480b9cb9605624a8439b3e14/projects/cm_mcu/MonitorI2CTask.h#L16-L20

Reply:
the issue of reading 4 or 12 bytes is that the register addresses for the 12-ch one are decreasing but the default i2c operation would read increasing register addresses. Though we can reorder outputs for 12-ch devices but this will add on to another exception.